bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -307,24 +293,9 @@ private void offer(Mutation mutation) throws IOException {
       Runnable operation = null;
       if (options.useBulkApi() && (mutation instanceof Put || mutation instanceof Delete)) {
         // TODO: Do this logic asynchronously.
-        synchronized (bulkMutationLock) {
-          if (bulkMutation == null) {
-            bulkMutation = new BulkMutation(this.adapter.getBigtableTableName().toString());
-          }
-          MutateRowRequest request = adapt(mutation);
-          SettableFuture<Empty> future = bulkMutation.add(request);
-          addExceptionCallback(future, mutation);
-          ListenableFuture<Empty> retryingFuture = asyncExecutor.addMutationRetry(future, request);
-
-          // Make sure that flush will not finish until the retries are finished.
-          rpcThrottler.registerRetry(retryingFuture);
-          addExceptionCallback(retryingFuture, mutation);
-          if (bulkMutation.getRowKeyCount() >= options.getBulkMaxRowKeyCount()
-              || bulkMutation.getApproximateByteSize() >= options.getBulkMaxRequestSize()) {
-            mutateRowsAsync(bulkMutation);
-            bulkMutation = null;
-          }
-        }
+        MutateRowRequest request = adapt(mutation);
+        ListenableFuture<Empty> future = bulkMutation.add(request);
+        addExceptionCallback(future, mutation);
       } else {
         initializeAsyncMutators();
```

Comment:
nit: I don't think you need to do this until you decide to add to the asyncOperationsQueue